zkwrapper/source/zkwrapper.cc

```diff
@@ -501,3 +542,37 @@ std::vector <uint8_t> ZKWrapper::get_byte_vector(const std::string &string) {
 void ZKWrapper::close() {
 	zookeeper_close(zh);
 }
+
+bool ZKWrapper::flush(const std::string& full_path, bool synchronous) const {
+	// flush is only blocking when the synchronous flag is set
+	if (synchronous) {
+
+		// I tried using condition variables, but it ended up failing on occasion, so sadly I resort to polling :(
+		bool * flag = (bool *)malloc(sizeof(bool));
+
+		// Lambda to call on function completion
+		string_completion_t completion = [](int rc, const char *value, const void *data){
+			bool * flag_ptr = const_cast<bool *>(reinterpret_cast<const bool *>(data));
+			* flag_ptr = true;
+			std::this_thread::sleep_for (std::chrono::milliseconds(1000));
+			free(flag_ptr);
+		};
+
+		int rc = zoo_async(zh, full_path.c_str(), completion, flag);
+
+		// Wait for the asynchronous function to complete
+		int count = 0;
+		// ZooKeeper is guaranteed to be synced within 2 seconds
+		while (!(* flag) && count < 2000) {
```

Comment:
Having the thread sleep for 1ms 2000 times is not necessarily 2s, I think. 
I think we can make the thread sleep for 2s and then just return flag.